File: swift/src/main/java/androidx/swift/session/SessionStateListener.java
package androidx.swift.session;

/**
 * 会话状态改变回调监听
 * 当用户信息发生改变时会触发该接口发出通知
 */
public interface SessionStateListener {

    /**
     * 用户信息改变
     * @param sessionManager 用户管理器
     */
    void onUserInfoChanged(SessionManager sessionManager);
}

File: kotlin/src/main/java/androidx/swift/session/SessionStateListener.kt
package androidx.swift.session

/**
 * Session状态监听
 * @author RAE
 * @date 2022/08/21
 * @copyright Copyright (c) https://github.com/raedev All rights reserved.
 */
interface SessionStateListener {

    /**
     * 用户信息发生改变
     * @param delegate Session管理器
     * @param oldUser 旧的用户
     * @param newUser 新的用户
     */
    fun <T : Any> onUserInfoChanged(delegate: SessionDelegate, oldUser: T?, newUser: T?)
}